### PR TITLE
fixed code from meeting

### PR DESCRIPTION
**Pull request by evan-schott**
Also noticed that start.go and debug.go have the same exact teardown() as cluster.go

### CODE REVIEW

**Comment by lthibault** [comment]
Yeah, the issue is that we're calling `Dial` within [Fx](https://pkg.go.dev/go.uber.org/fx) dependency-injection framework (see `newClientNode` in `internal/runtime/client.go`).  This means that any error will cause `fx.New()` to return a nil value for the `app` global.

I think the fix would be to populate a `client.Dialer`, which would then get called outside of Fx (namely: in the main `cli.ActionFunc`).

Want to take a crack at it?

I'll merge this in the meantime, and we can revert after we push a proper fix.

**Comment by evan-schott** [comment]
@lthibault I will take a stab at it

**Comment by evan-schott** [comment]
@lthibault 

My understanding is that Dial fails => newClientNode() fails => app = fx.New() fails => teardown() panics because it assumes app != null. 

So then since cluster does not have a cli.ActionFunc (unlike start), I should make one, and set up a client.Dialer in it so that way app != null, and then there is no panic at teardown? I am a little confused what the Dial actually does, and why it fails in that scenario (when cluster gets run without any running nodes).

**Comment by lthibault** [comment]
>My understanding is that Dial fails => newClientNode() fails => app = fx.New() fails => teardown() panics because it assumes app != null.

Correct 👍 

>So then since cluster does not have a cli.ActionFunc (unlike start), I should make one,

Not quite.  That would create an action for the `ww cluster` command (without any subcommand).

Instead, you should create a global `client.Dialer` and use `fx.Populate` to populate it, as we're [currently doing with `node`](https://github.com/wetware/ww/blob/master/internal/cmd/cluster/cluster.go#L69).

From there, you can call the `Dial` method within that same `cli.BeforeFunc`.  Make sure to remove the call to `Dial` from `internal/runtime/client.go`, too.

In short, we're moving the call to `Dial` from `internal/runtime/client.go` to `internal/cmd/cluster/cluster.go`, in the `cli.BeforeFunc`.

Does that make sense?
